tripy/tripy/frontend/utils.py

```diff
@@ -297,3 +297,35 @@ def topological_sort(ops: List[Union[BaseTraceOp, BaseFlatIROp]]) -> List[Union[
 
     assert len(ops) == len(result), f"Num original ops {len(ops)}, got num {len(result)}"
     return result
+
+
+# Processes a `dim` (i.e. axis) argument related to a tensor.
+# If the dimension is negative, this will convert it to the corresponding positive index.
+#
+# NOTE: This is currently an extremely specialized decorator that expects an `input` tensor
+# argument and a `dim` argument for the axis.
+# In the future, we can generalize this and use it more broadly.
+def process_dim(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        input = utils.get_arg_by_name("input", func, *args, **kwargs)
+
+        def process_dim(dim: int) -> int:
+            new_dim = dim
```

Comment:
Maybe add an error if dim is not python integer?

Reply:
We already have type checking in the frontend as a consequence of our function registry:
```
Could not find an implementation for function: 'repeat'.
    Note: Argument types were: [Tensor, int, dim=Tensor].
    Candidate overloads were:

    --> /tripy/tripy/frontend/utils.py:6 in repeat()
          |
        6 | @export.public_api(document_under="operations/functions")
        7 | @constraints.dtype_info(
        8 |     dtype_variables={
        9 |     ...
          | 

    Not a valid overload because: For parameter: 'dim', expected an instance of type: 'int' but got argument of type: 'Tensor'.
```